### PR TITLE
reconfirming running reservations can move available nodes

### PR DESCRIPTION
**Pull request by bhroam**
#### Describe Bug or Feature
When a running degraded reservation is reconfirmed, the scheduler  can replace all nodes, not just the unavailable ones.   This means the node a running job is on can be replaced.  This leaves the reservation with a job on a node that it doesn't own.

#### Describe Your Change
The main reason this happened is because when we were only reconfirming non-running reservations, we would mark the reservation as unconfirmed.  To the scheduler, a degraded reservation and an unconfirmed reservation were the same.  When we started reconfirming  running  reservations, this changed.  The problem is when we were choosing which select spec to use to confirm the reservation, we didn't detect the reservation was running.   We used the original select spec and treated the reservation like it was when it was not running.  This meant we could confirm it anywhere.  

I changed the code to leave  the reservation in the running state and have is_ok_to_run() detect it as confirmable in the degraded substate.  This meant check_nodes() sees it as running and we use the execselect.

#### Attach Test and Valgrind Logs/Output
[resv.log](https://github.com/openpbs/openpbs/files/4940516/resv.log)

### CODE REVIEW

File: src/scheduler/resource.c [hunk 1]
```diff
@@ -872,12 +872,12 @@ collect_resources_from_requests(resource_resv **resresv_arr)
...
 		 * that a job was run with qrun -H (res=val) where res is not part of
-		 * the schedselect.  The exec_vnode is taken directly from the -H  argument
+		 * the schedselect.  The exec_vnode is taken directly from the -H argument
 		 * The qrun -H case is why we need to do this check.
 		 */
 		if (r->execselect != NULL && r->execselect->defs != NULL) {
-			if ((r->is_job && r->job != NULL && in_runnable_state(r)) ||
-				(r->is_resv && r->resv != NULL && r->resv->resv_state == RESV_BEING_ALTERED)) {
+			if ((r->job != NULL && in_runnable_state(r)) ||
+				(r->resv != NULL && (r->resv->resv_state == RESV_BEING_ALTERED || r->resv->resv_substate == RESV_DEGRADED))) {
```
Comment (vstumpf) [inline src/scheduler/resource.c:880]: do you not need is_job or is_resv?

File: test/tests/functional/pbs_reservations.py [hunk 2]
```diff
@@ -1598,6 +1654,9 @@ def test_ASAP_resv_request_same_time(self):
         time doesn't crashes PBS daemon.
         """
         self.common_steps()
+        # turn opt_backfill_fuzzy so both reservations are at the same time
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'opt_backfill_fuzzy': 'off'}, id='default')
```
[the remark this answers is not in the document]
Comment (bhroam) [reply test/tests/functional/pbs_reservations.py:1659]: opt_backfill_fuzzy is unset  by default.  When it is unset, it is low.  I need to turn it completely off.

File: src/scheduler/resource.c (hunk 1 above)
Comment (bhroam) [reply src/scheduler/resource.c:880]: Yeah, I don't necessary need it, but unless we refactor is_job and is_resv out out of the code, I guess they should be used.  I'll remove them if you want.

**Comment by bhroam** [comment]
@vstumpf I addressed the typo.

File: test/tests/functional/pbs_reservations.py [hunk 3]
```diff
@@ -376,6 +376,62 @@ def test_degraded_advanced_reservation_superchunk(self):
         self.assertNotEqual(nds1, nds2)
         self.assertEquals(sc, nds1.split('+')[0])
 
+    def test_degraded_running_only_replace(self):
```
Comment (vstumpf) [inline test/tests/functional/pbs_reservations.py:379]: I think this needs to skip on cpuset machines

**Pull request merged**